app/models/job_template.rb

```diff
@@ -37,6 +37,28 @@ def self.base_class
   end
   self.table_name = 'templates'
 
+  # Import a template ERB, with metadata in the first YAML comment
+  def self.import(template, options = {})
+    return unless (match = template.match(/<%\#(.+?).-?%>/m)) && (metadata = YAML.load(match[1])) && metadata.delete('kind') == 'job_template'
+
+    if self.find_by_name(metadata['name']).blank?
+      inputs = metadata.delete('template_inputs')
+      template = self.create(metadata.merge(:template => template.gsub(/<%\#.+?.-?%>/m, '')))
+      template.update_attributes(options)
```

Comment:
This awkward dance is because you can't instantiate a new record with `:locked => true` in Foreman